### PR TITLE
imp!: use expected interface for legacy params subspace

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Allows passing `nil` to `NewKeeper` constructors for new chains which _do not_ need to migrate from `x/params`

closes: #4806 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/27-interchain-accounts/host/keeper/migrations.go [hunk 1]
```diff
@@ -21,8 +25,13 @@ func NewMigrator(k *Keeper) Migrator {
...
 func (m Migrator) MigrateParams(ctx sdk.Context) error {
 	if m.keeper != nil {
+		legacySubpsace, ok := m.keeper.legacySubspace.(paramtypes.Subspace)
+		if !ok {
+			return errorsmod.Wrapf(ibcerrors.ErrInvalidType, "expected %T, got %T", paramtypes.Subspace{}, m.keeper.legacySubspace)
+		}
+
 		var params types.Params
-		m.keeper.legacySubspace.GetParamSet(ctx, &params)
+		legacySubpsace.GetParamSet(ctx, &params)
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/host/keeper/migrations.go:34]: I can do that if you'd prefer. I essentially went with copy/pasta of the same interface that the sdk had used and doing a strict type conversion in the migrations so an error will be returned explicitly if needs be

[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/host/keeper/migrations.go:34]: On reflection, you get the same compiler safety by defining this as the expected interface func and not doing a strict type check 😅 I'll make the change

[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/host/keeper/migrations.go:34]: I think the only difference is that you'd get a runtime panic in migrations when passing `nil`, as opposed to a runtime error.

[the remark this answers is not in the document]
Comment (colin-axner) [reply modules/apps/27-interchain-accounts/host/keeper/migrations.go:34]: the code panics anyways on migration error

[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/host/keeper/migrations.go]: Nope, they were boilerpalte. Updated to use `GetParamSet` on the interface now.

**Review by colin-axner — APPROVED** [review]

**Review by charleenfei — APPROVED** [review]
thanks for the quick PR @damiannolan

**Pull request merged**